drivers/hmis/app/models/hmis/hud/custom_service_category.rb

```diff
@@ -18,6 +18,15 @@ class Hmis::Hud::CustomServiceCategory < Hmis::Hud::Base
 
   validates_presence_of :name, allow_blank: false
 
+  scope :non_hud, -> do
```

Comment:
Despite the left join, this will exclude empty categories because of the `IS NOT NULL` clause:
```
"SELECT \"CustomServiceCategories\".* FROM \"CustomServiceCategories\" LEFT OUTER JOIN \"CustomServiceTypes\" \"service_types\" ON \"service_types\".\"DateDeleted\" IS NULL AND \"service_types\".\"custom_service_category_id\" = \"CustomServiceCategories\".\"id\" WHERE \"CustomServiceCategories\".\"DateDeleted\" IS NULL AND \"service_types\".\"hud_record_type\" IS NOT NULL"
```

Maybe a better way would be to pick any custom service category that is empty or has any service with a null record_type. That would also make this match the "any" logic in the mutation:
```
irb(main):032:0> Hmis::Hud::CustomServiceCategory.left_joins(:service_types).where(service_types: { hud_record_type: nil }).to_sql
=> "SELECT \"CustomServiceCategories\".* FROM \"CustomServiceCategories\" LEFT OUTER JOIN \"CustomServiceTypes\" \"service_types\" ON \"service_types\".\"DateDeleted\" IS NULL AND \"service_types\".\"custom_service_category_id\" = \"CustomServiceCategories\".\"id\" WHERE \"CustomServiceCategories\".\"DateDeleted\" IS NULL AND \"service_types\".\"hud_record_type\" IS NULL"

```

Reply:
To make this suggestion a little clearer: it would support frontend users adding new customer service types to the "Check In" category example (which has both bed night and day shelter check-in). But it would not support adding customer services to the "stock" HUD service categories that are set up initially (that happens here https://github.com/greenriver/hmis-warehouse/blob/f06264d238d94eccc73ba50b6fe6b498c44efbdd/drivers/hmis/lib/hmis_util/service_types.rb#L17-L21)

Reply:
How did the "Check In" example get the way it is? Did it exist as a seeded category and then have a custom type added to it? The ui changes we are introducing in this PR would prevent more examples like that from cropping up, right? Or did we add it manually and configure it that way on purpose?

Reply:
It was set up manually in the rails console. I probably renamed the existing bed night category. Yes it was on purpose to have "bed night" and "check-in" in the same group, but I think it's fine to have the UI disallow that going forward. All the custom categories have been set up in the console since we don't have a UI for it yet.

Reply:
Thanks! Just to clarify,

> All the custom categories have been set up in the console since we don't have a UI for it yet

We don't have a UI for creating a custom category, but we have an existing UI for adding a custom type to a HUD category, thereby converting that category to a "Custom" category.
- Visit the Services admin UI, click New Service Type, and create a new custom service type
- Pick a HUD service category in the dropdown
- In the Categories tab, you can see that that category changed its "Type" column from "HUD Services" to "Custom Services." (In QA you can see that I just did that this morning with the "Bed Night" category when I added the new type "Custom bed night")

This is what I was asking about when I asked how the "Check In" example got the way it is -- did we do it intentionally on the console, or did the client do it (possibly unwittingly) using the UI following the steps I described above. Sounds like it was intentional on the console. 👍